plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchResponseProcessor.java

```diff
@@ -316,22 +321,37 @@ private void rewriteResponseDocuments(SearchResponse response, ActionListener<Se
      * @param inputMapIndex           the index of the input mapping to process
      * @param batchPredictionListener the listener to be notified when the predictions are processed
      * @param hitCountInPredictions   a map to keep track of the count of hits that have the required input fields for each round of prediction
+     * @param queryString             the query body in string format, for example, "{ \"query\": { \"match_all\": {} } }\n"
      * @throws IOException if an I/O error occurs during the prediction process
      */
     private void processPredictions(
         SearchHit[] hits,
         List<Map<String, String>> processInputMap,
         int inputMapIndex,
         GroupedActionListener<Map<Integer, MLOutput>> batchPredictionListener,
-        Map<Integer, Integer> hitCountInPredictions
+        Map<Integer, Integer> hitCountInPredictions,
+        String queryString
     ) throws IOException {
 
         Map<String, String> modelParameters = new HashMap<>();
         Map<String, String> modelConfigs = new HashMap<>();
 
         if (inferenceProcessorAttributes.getModelConfigMaps() != null) {
-            modelParameters.putAll(inferenceProcessorAttributes.getModelConfigMaps());
-            modelConfigs.putAll(inferenceProcessorAttributes.getModelConfigMaps());
+            Map<String, String> modelConfigMapsInput = inferenceProcessorAttributes.getModelConfigMaps();
+
+            for (Map.Entry<String, String> entry : modelConfigMapsInput.entrySet()) {
+                String modelConfigKey = entry.getKey();
+                String modelConfigValue = entry.getValue();
+                if (StringUtils.isValidJSONPath(modelConfigValue)) {
+                    Object querySubString = JsonPath.using(suppressExceptionConfiguration).parse(queryString).read(modelConfigValue);
+                    if (querySubString != null) {
```

Comment:
In the example if model config value is valid json path like `"query_text": "$.query.term.dairy.value"`, we will always parse the value, then this line checks if the value of the query path is null or not. If the value is null, we just do nothing, so the `query_text` will still be `"$.query.term.dairy.value"`, then the model will calculate similarity score of `"$.query.term.dairy.value"` and each doc, right ? This looks somehow hide such error and not easy to figure out

Reply:
let me add unit test for the querySubString is null when we  cannot find the query_text in the query string and test the error message

Reply:
Is `querySubString` the value of the query path ? The name seems confusing

Reply:
renamed to `queryText`